### PR TITLE
Add hide activity option

### LINKED ISSUE #7927: "Activities" tab and the logging for it is bad for the privacy - should be optional
- Gitea version (or commit ref): unknown (codeberg.org)
- Git version: not relevant
- Operating system: not relevant
- Database (use `[x]`): not relevant
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (<https://try.gitea.io/xandertest?tab=activity>)
  - [ ] No
  - [ ] Not relevant

## Description

Git logs the commit timestamps (which a developer can spoof for better privacy), however gitea logs the push timestamps and some other events. Is it possible to add the option to hide/ disable it per user/ project/ organization?

Note: In the linked example, there is the text "3 weeks ago". When moving the mouse over it, then there is the exact timestamp visible.

### PR DESCRIPTION
**Pull request by l-jonas**
This closes https://github.com/go-gitea/gitea/issues/7927

I was not able to setup drone, so I did not test according to <https://github.com/go-gitea/gitea/blob/master/CONTRIBUTING.md#testing-redux>

### CODE REVIEW

**Comment by 6543** [comment]
loged in none admins still see it too

**Comment by 6543** [comment]
Bildschirmfoto zu 2020-05-10 16-56-04

File: models/action.go [hunk 1]
```diff
@@ -319,6 +319,12 @@ func GetFeeds(opts GetFeedsOptions) ([]*Action, error) {
 		cond = cond.And(builder.In("repo_id", AccessibleRepoIDsQuery(opts.Actor)))
 	}
 
+	if opts.Actor == nil || !opts.Actor.IsAdmin {
```
Comment (6543) [inline models/action.go:322]:
```suggestion
```

File: models/action.go [hunk 2]
```diff
@@ -319,6 +319,12 @@ func GetFeeds(opts GetFeedsOptions) ([]*Action, error) {
...
+		if opts.RequestedUser.KeepActivityPrivate && actorID != opts.RequestedUser.ID {
+			return make([]*Action, 0), nil
+		}
+	}
```
Comment (6543) [inline models/action.go:326]:
```suggestion
```

**Review by 6543 — CHANGES_REQUESTED** [review]
.

File: models/action.go (hunk 2 above)
Comment (l-jonas) [reply models/action.go:326]: I think we are not talking about the same expected behaviour. It is intended that admins can still see it (as shown in the UI messages). That's what causes the condition. If the user is an admin, then the whole block which could return nothing should be skipped. If the user is not an admin, then the block should be executed (second condition). However, this throws an exception if there is no actor/ the requesting user is not signed in. Due to that, it is always executed if the actor is unknown. This skips the second condition check, always runs the block and prevents the crash.

**Comment by l-jonas** [comment]
Do I still need to run ``make fmt``? I think it would format the code, but I made changes according to the linter messages and the CI accepts it now.

The ``make fmt`` if it does what I think would be a good info for the Readme or the contribution guide. Should I open a pull request for that?

**Comment by 6543** [comment]
> Do I still need to run make fmt? I think it would format the code, but I made changes according to the linter messages and the CI accepts it now.
> 
> The make fmt if it does what I think would be a good info for the Readme or the contribution guide. Should I open a pull request for that?

sorry my tests somehow meesed up - now it works as it should so forget/resolve the suggestions

**Comment by 6543** [comment]
but if we speac about tests, it would be nice to add Tests for this

File: models/action.go (hunk 2 above)
Comment (6543) [reply models/action.go:326]: my test setup messed up `if opts.Actor == nil || !opts.Actor.IsAdmin {` was always returning false ?!?

just considder this as resolved

File: models/action.go (hunk 1 above)
Comment (6543) [reply models/action.go:322]: considder this as resolved

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -381,6 +381,7 @@ follow = Follow
 unfollow = Unfollow
 heatmap.loading = Loading Heatmap…
 user_bio = Biography
+disabled_public_activity = This user has disabled the public visibility of the activity. Only the user itself and admins can see it.
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
disabled_public_activity = This user has disabled the public visibility of the activity.
```

Comment (6543) [reply options/locale/locale_en-US.ini]: a normal user dont need the hint ...

**Review by 6543 — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
Thanks for this PR :)

I hope you don't mind that I've resolved the conflicts.

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by 6543** [comment]
Labled as feature to take more atraction to this option - it would go lost in the huge changelog
And this is something people activly should decide

**Pull request merged**